### PR TITLE
chore(multitenancy): v2 api isolation lesson template (#6401)

### PR DESCRIPTION
**Pull request by corinnekrych**
### Proposed changes

*
*

### Testing Instructions

1. Step-by-step how to test
2. Environment or config notes

### Related issues

* Closes #ISSUE-NUMBER
*

### Checklist

- [ ] I consider the submitted work as finished
- [ ] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality
- [ ] For bug fix -> I implemented a test that covers the bug

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: .github/skills/activate-tenant-table/SKILL.md [hunk 1]
```diff
@@ -376,32 +377,94 @@ Rebase right before go-live and re-read the merged allowlist line; a bad
...
+
+**Classification and action:**
+
+| Finding | On `{table}`? | Action |
+|---------|---------------|--------|
+| `TenantContext.getCurrentTenant()` used to look up `{Entity}` | Yes | **Remove** — the inspector scopes the query |
+| `TenantContext.getCurrentTenant()` used to look up a *different* entity | No | **Report only** — out of scope for this activation |
+| `findByIdAndTenantId` on `{EntityRepository}` | Yes | **Replace with `findById`** — inspector handles scoping, explicit tenant filter blocks multi-tenant reads |
+| `findByIdAndTenantId` on a *different* repository | No | **Report only** — that table is still on v1 |
+| `TenantBaseListener` on `{Entity}` | Yes | **Remove from `@EntityListeners`** — write attribution is now explicit |
```
Comment (laugiov) [inline .github/skills/activate-tenant-table/SKILL.md]:
```suggestion
| `TenantBaseListener` on `{Entity}` | Yes | **Remove only if every write path attributes the tenant explicitly** (resolver / `setTenant`); otherwise **keep** — a listener-dependent path (e.g. an importer that `save()`s without `setTenant`) would write a NULL tenant |
```
I verified this on kill_chain_phases: its import (V1_DataImporter.createKillChainPhase) saves without setTenant, so removing the listener writes a NULL tenant. 
For lessons_templates it's safe (only createLessonsTemplate writes, and it attributes explicitly). 
Note the file currently contradicts itself, line 312 says "Leave TenantBaseListener" while the go-live step (line 349) and this row say "remove"; worth aligning all three to the conditional rule.

File: .github/skills/activate-tenant-table/SKILL.md [hunk 2]
```diff
@@ -376,32 +377,94 @@ Rebase right before go-live and re-read the merged allowlist line; a bad
...
 catches, do not rely on it alone.
 
-### Phase 7 — Full regression pass
+### Phase 7 — v1 remnant audit
+
+Before running the regression suite, audit the full call stack of every public
+method on `{Api}` (and on the other APIs from Phase 5) for v1 isolation
+patterns that conflict with or duplicate the v2 mechanism.
+
+**What to search for:**
```
Comment (laugiov) [inline .github/skills/activate-tenant-table/SKILL.md:410]: Different concern from v1 remnants, but same activation impact and worth a companion check (here or in the inventory phase): adding a TxCtx parameter to an endpoint breaks tests a repository grep misses 

- a standaloneSetup / @WebMvcTest MockMvc test hitting the URL returns 500 (No primary or single unique constructor found for interface TxCtx), that setup has no TxCtxArgumentResolver;
- a test calling the controller method directly in Java breaks at compile time.

I hit both (OpenCTIApiTest, KillChainPhaseApiTest). Scan:

- grep -rln "standaloneSetup" openaev-api/src/test/java | xargs grep -l "{Api}\|/{entities}"
- grep -rn "{Api}" openaev-api/src/test/java   # direct calls

Fix: register the resolver on the standalone builder (.setCustomArgumentResolvers(new TxCtxTestArgumentResolver(...))) or move to the full-context IntegrationTest base; add the TxCtx arg to direct calls.

File: openaev-api/src/test/java/io/openaev/rest/lessons_template/LessonsTemplateHttpIsolationTest.java [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+            .andExpect(status().isOk())
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+    assertTrue(response.contains(templateA), "A's template must appear in A's search results");
+    assertFalse(response.contains(templateB), "B's template must not appear in A's search results");
+  }
+
+  @Test
+  @Disabled(
```
Comment (laugiov) [inline openaev-api/src/test/java/io/openaev/rest/lessons_template/LessonsTemplateHttpIsolationTest.java]: This case is disabled because the X-Tenant-Ids header selector returns an empty set. 
On the import_mappers pilot this exact case was re-enabled at go-live and passes, so it's worth confirming, before merge, whether the header selector is genuinely broken for lessons_templates or the test setup differs from the pilot's. 
Any idea about the root cause ?

Comment (laugiov) [reply openaev-api/src/test/java/io/openaev/rest/lessons_template/LessonsTemplateHttpIsolationTest.java]: Update: I couldn't reproduce the empty result , it passes both in isolation and within the full LessonsTemplateHttpIsolationTest on a clean DB. 
So it doesn't look like the header selector is broken; more likely a stale local DB or cross-class test state when you saw it empty. 
Could you try re-running it on a fresh DB? 
And if it only fails inside the full API suite, that points to a test-isolation issue rather than a product bug.

Comment (corinnekrych) [reply openaev-api/src/test/java/io/openaev/rest/lessons_template/LessonsTemplateHttpIsolationTest.java]: fixed a5b3ccaa5

I think AI miss the removal of @Disabled. we should remove thsoe step https://github.com/OpenAEV-Platform/openaev/blob/4fd373f9088f0599597675e66833b42195df5fe4/.github/skills/activate-tenant-table/SKILL.md#L248-L252 and keep it simple no temporary @disable

File: .github/skills/activate-tenant-table/SKILL.md (hunk 2 above)
Comment (corinnekrych) [reply .github/skills/activate-tenant-table/SKILL.md:410]: fixed with 776684eb2
I've added your comment

we can refine skills further in next PR if needed

**Review by laugiov — APPROVED** [review]
